### PR TITLE
Workflow hierarchy

### PR DESCRIPTION
**Pull request by NicoDeVeaux**
Creates a bbsr_tfa workflow that the bsubtilis case uses directly, and the yeast workflow inherits from. 

This is still a work in progress because we might decide to use the filtering from the yeast workflow in the base, but wanted to get feedback so created a pull request

### CODE REVIEW

File: inferelator_ng/yeast_bbsr_workflow.py [hunk 1]
```diff
@@ -6,70 +6,18 @@
...
         """
-        Output result report(s) for workflow run.
+        Output result report(s) for workflow run, using a binary version of the gold standard
         """
-        output_dir = os.path.join(self.input_dir, datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S'))
-        os.makedirs(output_dir)
-        self.results_processor = ResultsProcessor(betas, rescaled_betas)
-        self.results_processor.summarize_network(output_dir, abs(gold_standard), priors)
-
+        return super(Yeast_Bbsr_Workflow, self).emit_results(betas, rescaled_betas, abs(gold_standard), priors)
```
Comment (NicoDeVeaux) [inline inferelator_ng/yeast_bbsr_workflow.py]: @kostyat @dayanne-castro This is a way we can very tersely modify the yeast workflow to use the absolute value of the gold standard while still using the logic of `emit_results` in the parent class

Comment (dayanne-castro) [reply inferelator_ng/yeast_bbsr_workflow.py]: Don't we usually run with the absolute value? I didn't even realize that one.

**Review by dayanne-castro — COMMENTED** [review]
I like this much better, way more compact. I see that you also changed the filter_expression_and_priors. To me, that's how it should be done and we should just change the unit test.

File: inferelator_ng/yeast_bbsr_workflow.py (hunk 1 above)
Comment (NicoDeVeaux) [reply inferelator_ng/yeast_bbsr_workflow.py]: The bsubtilis gold standard has only positive values (0 or 1). So there was no need to add an absolute value function before checking against it. 
The only gold standard with negative values is the yeast one.

**Comment by NicoDeVeaux** [comment]
Hi @dayanne-castro , 

I wrote a test for a negative gold standard and things totally broke: the results processor as it is currently implemented takes the cumulative sum of the GS, so negative GS values can lead to negative precisions and recalls!

For example, in the new test test_negative_gs_aupr_perfect_prediction, the output AUPR is -1! 

In my mind, there are two options:

a) Re-write results processor so that it matches negative entries in GS to negative predictions
 
OR

b) Cast the GS to a positive binary matrix in the resutls processor itself, so that it never can get into the position of emitting negative AUPRs, precisions, etc. 

I'm in favor of b), because there isn't yet a use case for a).

Nick

**Comment by dayanne-castro** [comment]
I thought about it and I agree with changing the gold standard to a 1,0 matrix in results processor. Up until now this is how we have done anyway, evaluating structure and not signs.

Thanks, Nick!

File: inferelator_ng/yeast_bbsr_workflow.py (hunk 1 above)
Comment (ryi06) [reply inferelator_ng/yeast_bbsr_workflow.py]: We've decided that the Bsubtilis gold standard should have both negative and positive values. I'll update a new gold standard matrix and update PR curve calculation.

**Review by ryi06 — APPROVED** [review]

**Pull request merged**